### ISSUE TITLE
dh group exchange request - invalid buffer size 

### ISSUE THREAD

**Issue by GrooveCommand** [mscdex/ssh2-streams#85]
I recently came across an issue after an update was made to the ssh server I'm working with. I started receiving the following error when attempting to perform sftp.readdir()

```
events.js:182
      throw er; // Unhandled 'error' event
      ^

Error: The Transport Protocol thread failed:java.lang.NullPointerException
    at onDISCONNECT (C:\Support Tool\development\node_modules\ssh2-streams\lib\ssh.js:2202:15)
    at SSH2Stream.<anonymous> (C:\Support Tool\development\node_modules\ssh2-streams\lib\ssh.js:204:5)
    at emitMany (events.js:146:13)
    at SSH2Stream.emit (events.js:223:7)
    at parsePacket (C:\Support Tool\development\node_modules\ssh2-streams\lib\ssh.js:3748:10)
    at SSH2Stream._transform (C:\Support Tool\development\node_modules\ssh2-streams\lib\ssh.js:669:13)
    at SSH2Stream.Transform._read (_stream_transform.js:186:10)
    at SSH2Stream._read (C:\Support Tool\development\node_modules\ssh2-streams\lib\ssh.js:252:15)
    at SSH2Stream.Transform._write (_stream_transform.js:174:12)
    at doWrite (_stream_writable.js:387:12)
```

Upon inspecting a packet capture I could see that the Diffie-Hellman Group Exchange Request was triggering a disconnect from the SSH Server. Here's what Wireshark showed for the Diffie-Hellman Group Exchange Request:

```
Key Exchange
    Message Code: Diffie-Hellman Group Exchange Request (34)
    DH GEX Min: 1024
    DH GEX Number of Bits: 2560
    DH GEX Max: 5120
    Padding String: 99c2df0babd2
```

Long story short, I was able to resolve the issue by correcting the buffers defined in ssh.js (line 85) with the following:

```
var KEXDH_GEX_REQ_PACKET = new Buffer([
  MESSAGE.KEXDH_GEX_REQUEST,
  // Minimal size in bits of an acceptable group
  0, 0, 4, 0, // 1024, modp2
  // Preferred size in bits of the group the server will send
  0, 0, 16, 0, // 4096, modp16
  // Maximal size in bits of an acceptable group
  0, 0, 32, 0 // 8192, modp18
]);
```

After making the above change, the Diffie-Hellman Group Exchange Request showed the correct values:

```
Key Exchange
    Message Code: Diffie-Hellman Group Exchange Request (34)
    DH GEX Min: 1024
    DH GEX Number of Bits: 4096
    DH GEX Max: 8192
    Padding String: 99c2df0babd2
```
